File: server/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/IssueReportStatusUpdateRequest.java
package edu.cnm.deepdive.seesomethingabq.model.dto;

/**
 * Request body DTO for updating the accepted-state of an issue report.
 */
public class IssueReportStatusUpdateRequest {

  private String statusTag;

  /**
   * Returns the new accepted-state status tag.
   *
   * @return status tag.
   */
  public String getStatusTag() {
    return statusTag;
  }

  /**
   * Sets the new accepted-state status tag.
   *
   * @param statusTag status tag.
   */
  public void setStatusTag(String statusTag) {
    this.statusTag = statusTag;
  }

}

File: app/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/IssueReportStatusUpdateRequest.kt
package edu.cnm.deepdive.seesomethingabq.model.dto

import com.google.gson.annotations.SerializedName

/**
 * Request DTO for updating the accepted-state/status of an issue report (manager-only).
 *
 * @property statusTag desired status tag.
 */
data class IssueReportStatusUpdateRequest(
  @SerializedName("statusTag")
  val statusTag: String
)

